libr/core/canal.c

```diff
@@ -4766,29 +4766,50 @@ static bool __cb(RFlagItem *fi, void *user) {
 static int __addrs_cmp(void *_a, void *_b) {
 	ut64 a = r_num_get (NULL, _a);
 	ut64 b = r_num_get (NULL, _b);
-        return a - b;
+	if (a > b) {
+		return 1;
+	}
+	if (a < b) {
+		return -1;
+	}
+        return 0;
 }
 
+#define MAXFCNSIZE 1024*1024*1
 R_API void r_core_anal_inflags(RCore *core, const char *glob) {
 	RList *addrs = r_list_newf (free);
 	RListIter *iter;
+	bool a2f = r_config_get_i (core->config, "anal.a2f");
 	char *anal_in = strdup (r_config_get (core->config, "anal.in"));
 	r_config_set (core->config, "anal.in", "block");
+	// aaFa = use a2f instead of af+
+	bool simple = (glob && *glob == 'a')? false: true;
 	glob = r_str_trim_ro (glob);
-	bool simple = 1;
 	char *addr;
 	r_flag_foreach_glob (core->flags, glob, __cb, addrs);
+	// should be sorted already 
 	r_list_sort (addrs, (RListComparator)__addrs_cmp);
 	r_list_foreach (addrs, iter, addr) {
 		if (!iter->n || r_cons_is_breaked ()) {
 			break;
 		}
 		char *addr2 = iter->n->data;
+		if (!addr || !addr2) {
+			break;
+		}
 		ut64 a0 = r_num_get (NULL, addr);
 		ut64 a1 = r_num_get (NULL, addr2);
-		int sz = a1 - a0;
-		if (sz < 1) {
-			eprintf ("Warning: unsorted flag list\n");
+		if (a0 == a1) {
+			// ignore
+			continue;
+		}
+		if (a0 > a1) {
+			eprintf ("Warning: unsorted flag list 0x%llx 0x%llx\n", a0, a1);
+			continue;
+		}
+		st64 sz = a1 - a0;
+		if (sz < 1 || sz > MAXFCNSIZE) {
```

Comment:
why `MAXFCNSIZE` should be a limit? what's the issue in defining a function which is 1MB far from the last flag?

Reply:
because some libc and sometimes kernels will decide to crash if you alloc 8 TB of ram instead of just tell you that thats probably wrong in your code